clippy_lints/src/methods/mod.rs

```diff
@@ -264,6 +264,8 @@ declare_clippy_lint! {
     /// The method signature is controlled by the trait and often `&self` is required for all types that implement the trait
     /// (see e.g. the `std::string::ToString` trait).
     ///
+    /// Clippy allow `Pin<&Self>` and `Pin<&mut Self>` if `&self` and `&mut self` is required.
```

Comment:
NIT:
```suggestion
    /// Clippy allows `Pin<&Self>` and `Pin<&mut Self>` if `&self` and `&mut self` is required.
```
I would maybe write it like this:
```suggestion
    /// The lint allows self references to be wrapped inside
    /// [`Pin<>`](https://doc.rust-lang.org/std/pin/struct.Pin.html)
```
However, I'm not sure if that is more clear than your documentation. Do you have any thoughts on this? :upside_down_face:

Reply:
Personally speaking, I prefer my version.